### PR TITLE
Assingment 4

### LINKED ISSUE #118: Basic authentication
Using the `hapi-auth-basic` plugin, add a new `/private` endpoint protected by a username and password that returns a simple HTML page with a welcome message. The welcome message should include the logged in username. Do not use any templates (views) at this stage. For username/password validation, create a static `lib/users.json` file (we will replace this in the near future) that contains a few sample users and passwords.

The new endpoint will be contained in its own plugin, similar to how we set up `/version`, and will require bringing test coverage back to 100%.

Due: 4/27

### PR DESCRIPTION
**Pull request by AdriVanHoudt**
closes #118 

* I added the `v` option to test command because it's easier to see what is going wrong with your tests and since this is a learning thing I thought it might help (it helped me)
* Removed `describe` shortcut in test/index.js
* Renamed `Server` to `Hueniversity` test/version.js
    *  see https://github.com/hueniverse/hueniversity/commit/13b9f07922ab36fe4dde0479638d79baf6f099e5 for more explanation
* Added linting thresholds to make the tests fail when linting returns errors or warnings

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 test:
 	@node node_modules/lab/bin/lab -a code
 test-cov:
-	@node node_modules/lab/bin/lab -a code -t 100 -L
+	@node node_modules/lab/bin/lab -a code -t 100 -Lv --lint-errors-threshold 0 --lint-warnings-threshold 0
```
Comment (hueniverse) [inline Makefile]: Are these extra flags really needed?

Comment (AdriVanHoudt) [inline Makefile]: Debatable but Imo they will help prevent that the majority of the comments
on pr's are about following the hapi style.

File: lib/users.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+	"Foo": {
+		"username": "Foo",
```
Comment (hueniverse) [inline lib/users.json]: Safer to add the username programatically than repeat information.

File: test/index.js [hunk 3]
```diff
@@ -50,7 +49,7 @@ it('handles register plugin errors', { parallel: false }, function (done) {
         name: 'fake version'
     };
 
-    Hueniversity.init(0, function (err, server) {
```
Comment (hueniverse) [inline test/index.js]: Better to keep the callback function with all its actual arguments. Kind of self documenting.

File: lib/users.json (hunk 2 above)
Comment (AdriVanHoudt) [inline lib/users.json]: you mean not have the username and assume the key is?

File: test/index.js (hunk 3 above)
Comment (AdriVanHoudt) [inline test/index.js]: Seems like a waste to pass around unused objects, the callback from the init function is self documenting enough imo, but you have a point

**Pull request merged**